service/src/main/java/com/excilys/service/importTool/impl/ComputerImportService.java

```
@@ -0,0 +1,152 @@
+            file.transferTo(convFile);
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        try {
+            FileReader fr = new FileReader(convFile);
+            CSVReader csvReader = new CSVReader(fr, ',');
+            csvReader.readNext();
+            String[] nextLine;
+
+            while ((nextLine = csvReader.readNext()) != null) {
+                String name = nextLine[0];
+                String introducedString = nextLine[1];
+                String discontinuedString = nextLine[2];
+                String companyName = nextLine[3];
+
+                ComputerDTO computerDTO = new ComputerDTO.Builder()
+                        .name(name)
+                        .introduced(introducedString)
+                        .discontinued(discontinuedString)
+                        .companyName(companyName)
+                        .build();
+
+                Map<Fields, List<ErrorMessage>> errors = computerValidator.validateComputerDTO(computerDTO);
+                Error conflict = new Error();
+                if (errors.size() != 0) {
+                    conflict.setComputerDTO(computerDTO);
+                    conflict.setErrorMap(errors);
+                    rapport.getRefuse().add(conflict);
+                } else {
+                    rapport.getToImport().add(computerDTO);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return rapport;
+    }
+
+    @Override
+    public Rapport importComputersFromXML(MultipartFile file) {
+        File convFile = new File(file.getOriginalFilename());
+        try {
+            file.transferTo(convFile);
+        } catch (IllegalStateException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
```

Idem (Throw a runtimeException or ignore the exception)